moonshot/integrations/cli/benchmark/recipe.py

```diff
@@ -329,10 +331,19 @@ def run_recipe(args) -> None:
         ):
             raise TypeError(ERROR_BENCHMARK_RUN_RECIPE_ENDPOINTS_VALIDATION)
 
-        if isinstance(args.num_of_prompts, bool) or not isinstance(
-            args.num_of_prompts, int
+        if isinstance(args.prompt_selection_percentage, bool) or not isinstance(
+            args.prompt_selection_percentage, int
+        ):
```

Comment:
is the bool check required?

Reply:
yup you are right. it is not needed.

Reply:
This bool check is needed, this is because in the unit test where the prompt_selection_percentage argument is True, it will be considered as an integer of 1. 

Refer to test_recipe line 2009 to 2019 for the test case
`(
    "Test Runner",
    "['recipe1', 'recipe2']",
    "['endpoint1', 'endpoint2']",
    10,
    42,
    "Test system prompt",
    True,
    "result_module",
    "[run_recipe]: The 'runner_proc_module' argument must be a non-empty string and not None.",
)`


`
FAILED tests/unit-tests/cli/test_recipe.py::TestCollectionCliRecipe::test_run_recipe[Test Runner-['recipe1', 'recipe2']-['endpoint1', 'endpoint2']-True-42-Test system prompt-runner_module-result_module-[run_recipe]: The 'prompt_selection_percentage' argument must be an integer.] - AssertionError: assert '[run_recipe]...e an integer.'  1 failed, 2251 passed, 105 warnings in 30.35s `